bloom_nofos/nofos/nofo.py

```diff
@@ -234,6 +234,84 @@ def create_nofo(title, sections):
         raise e
 
 
+def preserve_subsection_metadata(nofo, new_sections):
+    preserved_metadata = {}
+    callout_count = 0
+    html_class_count = 0
+    callout_items = []
+    html_class_items = []
+
+    # Create lookup dict for new sections
+    new_section_lookup = {}
+    for section in new_sections:
+        section_name = section.get("name", "")
+        for subsection in section.get("subsections", []):
+            if not subsection.get("name"):
+                continue
+
+            key = f"{section_name}|{subsection.get('name')}"
```

Comment:
Subsections don't need to have names (for example subsections that come immediately after a callout box often don't have a `.name`), so we need to think of something to match on if there is no section name. 

Maybe we look at the first line/sentence of body content?